core/block/export/export.go

```diff
@@ -620,54 +645,142 @@ func (e *export) cleanupFile(wr writer) {
 }
 
 func (e *export) getRelatedDerivedObjects(objects map[string]*types.Struct) ([]database.Record, error) {
+	derivedObjects, err := e.iterateObjects(objects)
+	if err != nil {
+		return nil, err
+	}
+	if len(derivedObjects) > 0 {
+		derivedObjectsMap := make(map[string]*types.Struct, 0)
+		for _, object := range derivedObjects {
+			id := object.Get(bundle.RelationKeyId.String()).GetStringValue()
+			derivedObjectsMap[id] = object.Details
+		}
+		iteratedObjects, err := e.iterateObjects(derivedObjectsMap)
+		if err != nil {
+			return nil, err
+		}
+		derivedObjects = append(derivedObjects, iteratedObjects...)
+	}
+	return derivedObjects, nil
+}
+
+func (e *export) iterateObjects(objects map[string]*types.Struct) ([]database.Record, error) {
 	var (
 		derivedObjects []database.Record
+		relations      []string
 		err            error
-		relationLinks  pbtypes.RelationLinks
 	)
-
 	for id, object := range objects {
-		err = getblock.Do(e.picker, id, func(b sb.SmartBlock) error {
+		err := getblock.Do(e.picker, id, func(b sb.SmartBlock) error {
 			state := b.NewState()
-			relationLinks = state.GetRelationLinks()
+			relations = e.getObjectRelations(state, relations)
+			details := state.Details()
+			if e.objectWithDataview(details) {
+				relations, err = e.getDataviewRelations(state, relations)
```

Comment:
It seems that we need `lo.Uniq` here. Maybe we should not pass already collected relations to `getDataviewRelations` func, but to merge return value with `relations` slice?